modules/support_ticket/support_ticket.module

```diff
@@ -253,6 +244,144 @@ function support_ticket_load($nid = NULL, $reset = FALSE) {
+    $config = \Drupal::configFactory()->get('support_ticket.settings');
+    $filter_format = $config->get('support_ticket_type_settings.' . $type->id() . '.filter_format');
+    if (!$view_mode = $config->get('support_ticket_type_settings.' . $type->id() . '.view_mode')) {
+      $view_mode = 'default';
+    }
+
+    // Get a RevisionController to generate a diff.
+    $diffFormatter = \Drupal::service('diff.diff.formatter');
+    $dateFormatter = \Drupal::service('date.formatter');
+    $pluginManager = \Drupal::service('plugin.manager.diff.builder');
+    $diffEntityParser = \Drupal::service('diff.entity_parser');
+    $revisionController = new \Drupal\support_ticket\Controller\SupportTicketRevisionController($diffFormatter, $dateFormatter, $pluginManager, $diffEntityParser);
+
+    // Compare the original entity with the current revision.
+    $original = $support_ticket->original;
+    $fields = $revisionController->compareRevisions($original, $support_ticket);
+    $base_fields = \Drupal::entityManager()->getBaseFieldDefinitions('support_ticket');
+    foreach ($fields as $name => $field) {
+      $visible = entity_get_display('support_ticket', $support_ticket->getType(), $view_mode)->getComponent($name);
+      if ($visible === NULL && !array_key_exists($name, $base_fields)) {
+        unset($fields[$name]);
+      }
+    }
+
+    $diff_rows = array();
+    // Format the differences as rows in a table.
+    foreach ($fields as $field) {
+      $field_label_row = '';
+      if (!empty($field['#name'])) {
+        $field_label_row = array(
+          'data' => t('Changes to %name', array('%name' => $field['#name'])),
+          'colspan' => 4,
+          'class' => array('field-name'),
+        );
+      }
+      $field_diff_rows = _support_ticket_get_rows(
+        $field['#states']['raw']['#left'],
+        $field['#states']['raw']['#right']
+      );
+
+      // Add the field label to the table only if there are changes to that field.
+      if (!empty($field_diff_rows) && !empty($field_label_row)) {
+        $diff_rows[] = array($field_label_row);
+      }
+
+      // Add field diff rows to the table rows.
+      $diff_rows = array_merge($diff_rows, $field_diff_rows);
+
+    }
+
+    // Create our render array.
+    $build['diff'] = array(
+      '#type' => 'table',
+      '#rows' => $diff_rows,
+      '#empty' => t('No visible changes'),
+      '#attributes' => array(
+        'class' => array('diff'),
+      ),
+    );
+
+    // Store generated diff as a comment.
+    $field_storage = FieldStorageConfig::loadByName('support_ticket', $field_name);
+    $comment_type = $field_storage->getSetting('comment_type');
+    $diff_comment = array(
+      'cid' => NULL,
+      'entity_id' => $support_ticket->id(),
+      'entity_type' => 'support_ticket',
+      'field_name' => $field_name,
+      'pid' => 0,
+      'comment_body' => array(
+        'value' => drupal_render($build['diff']),
```

Comment:
Okay, in the last pass-through I found something. You should use \Drupal::service('renderer')->renderPlain() here, because:

a) You don't care about cacheability metadata or assets at this point
b) This could be called from within a queue or cron or somewhere else and then drupal_render() fails, because there is no render context to render on.